doc/specs/stdlib_version.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Version information
+---
+
+# The `stdlib_version` module
+
+[TOC]
+
+## Introduction
+
+The `stdlib_version` module contains the version of the standard library.
+The version information can be used as compile time constant or retrieved from a getter function at runtime.
+In case the standard library is dynamically linked, the version number retrieved from the getter might mismatch the compile time constants provided from the version built against.
+Therefore, it is recommended to retrieve the version information always at runtime.
+
+
+## Constants provided by `stdlib_version`
+
+### `stdlib_version_string`
+
+String constant representing the version number.
+
+### `stdlib_version_compact`
+
+Compact representation of the version string following the scheme:
+major * 10000 + minor * 100 + patch.
+
+
+### `get_stdlib_version`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Getter function to retrieve version information
+
+#### Syntax
+
+`res = [[stdlib_version(module):get_stdlib_version(function)]] ([major], [minor], [patch], [string])`
+
+#### Class
+
+Pure subroutine.
+
+#### Argument
+
+`major`: shall be an intrinsic integer type. It is an optional, `intent(out)` argument.
+`minor`: shall be an intrinsic integer type. It is an optional, `intent(out)` argument.
+`patch`: shall be an intrinsic integer type. It is an optional, `intent(out)` argument.
+`string`: shall be a deferred length character type. It is an optional, `intent(out)` argument.
```

Comment:
Just wondering if a fixed-length string can also be used to collect the output?

Reply:
The dummy argument is currently `character(len=:), allocatable, intent(out), optional`, since we are using a subroutine rather than a function. The rationale is that we want a stable API *and* ABI for this procedure, we could also go for multiple functions for getting the individual values, to make use of it in a more functional way.